libs/entitlement/qa/entitlement-nonmodular/src/main/java/org/elasticsearch/test/entitlements/RestEntitlementsCheckSystemExitAction.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the "Elastic License
+ * 2.0", the "GNU Affero General Public License v3.0 only", and the "Server Side
+ * Public License v 1"; you may not use this file except in compliance with, at
+ * your election, the "Elastic License 2.0", the "GNU Affero General Public
+ * License v3.0 only", or the "Server Side Public License, v 1".
+ */
+
+package org.elasticsearch.test.entitlements;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.elasticsearch.client.internal.node.NodeClient;
+import org.elasticsearch.rest.BaseRestHandler;
+import org.elasticsearch.rest.RestRequest;
+
+import java.util.List;
+
+import static org.elasticsearch.rest.RestRequest.Method.GET;
+
+public class RestEntitlementsCheckSystemExitAction extends BaseRestHandler {
+
+    private static final Logger logger = LogManager.getLogger(RestEntitlementsCheckSystemExitAction.class);
+
+    RestEntitlementsCheckSystemExitAction() {}
+
+    @Override
+    public List<Route> routes() {
+        return List.of(new Route(GET, "/_entitlement/nonmodular/_check_system_exit"));
```

Comment:
I wonder if we could make the nonmodular case more robust by having nonmodular plugin for both allowed and denied, and making them extend the appropriate modular plugin. It can register the same rest handler (it would just need a single Plugin class to register that), but this nonmodular plugin could then reuse almost everything so we have less boilerplate needed for each check test.

Reply:
Parametrized tests use the set of actions known to the handler _combined_ with 2 endpoint prefixes, and run all the actions on the 2 endpoints, which resolve to modular/non-modular plugin implementations.